### DISCUSSION TITLE
RGT not writing data in logs

### DISCUSSION THREAD

**Discussion by GShock112** [mytourbook/mytourbook#1059] [Ideas]
And so we found out here https://github.com/mytourbook/mytourbook/discussions/1046 that the RGT devs do not write down calories in their fit/gpx files. 

As I go forward to discorver their program and upload the logs into RGTCycling (app for tacx indoor trainer), I managed to use a calculator https://www.omnicalculator.com/sports/calories-burned-by-heart-rate and by editing the tour I can insert them easily. What I did find out this time is that the power data is present on the fit file but not the gpx file whereas the gpx contains altimetric data of some sort.

Unfortunately, there is no way to insert manually these data into MTB. So with the GPX the max pwr, normalized pwr etc is not present (but it's in the fit file) and other data missing from the fit file is present in the gpx file. 

I can't find a way to MERGE these 2 logs so that MTB would use all data but maybe it's me... MTB is very complex. Is there a way? 

In any case, it would be nice to be able to input any data in any of the stat columns you can see. I could input the data missing from either log into the other, then delete the duplicate. 

We could use an indoor input at the weather type when editing the tour (I am leaving it blank for the indoor training but I would rather input INDOORS as an option present with all the other type of weather) 

Ah btw, you can see the data missing from one log and the other: https://we.tl/t-IBSs9S8mas
Putting one next to the other it's easy to spot what one log writes and what the other doesn't. 

Thanks guys, I love MTB... every time I resume training I learn something new about this program. Hopefully I won't get injured anytime soon and stop again. :-)))

**Comment by FJBDev** [comment]
> I can't find a way to MERGE these 2 logs so that MTB would use all data but maybe it's me... MTB is very complex. Is there a way?
> 

If you read the MT's documentation you will find that there is a way to [merge tours](https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/tools/merge-tours) since several years now.


> Ah btw, you can see the data missing from one log and the other: https://we.tl/t-IBSs9S8mas Putting one next to the other it's easy to spot what one log writes and what the other doesn't.
> 
Actually, when I take your files and use the existing merge funtion, this is what I can merge from the GPX to the FIT file

![image](https://user-images.githubusercontent.com/25235665/230945041-4480af5f-b926-4a43-b129-fe3d49b9c586.png)


![image](https://user-images.githubusercontent.com/25235665/230944726-7166c538-3266-4ab1-9dcd-1d93e04a16cb.png)

**Comment by wolfgang-ch** [comment]
It's interresting, that this almost 15 year old tool is still useful. I implemented it because my first GPS device (Garmin Edge 705) had no temperature but my none GPS device had temperature. So I used this tool for several years until I bought a GPS device with temperature.



![image](https://user-images.githubusercontent.com/1283445/230953336-5f5807f9-1407-46e1-b015-b03adc7c0295.png)

![image](https://user-images.githubusercontent.com/1283445/230952193-5bb2e1a4-e856-4d82-94e7-34cdad85428a.png)

**Comment by GShock112** [comment]
Yes, at any rate RGT has a long way to go before producing a fully functional log but the program delivers both files and each is missing something the other one has, therefore the merge tool will solve the problem, yes I see it now. My next challenge will be to cut a log at its final part. The program allows you the free ride function but when you exit you don't get an option to close the log. I am very sure this function is already in MTB. 

As of the calories entirely missing, I am using this tool: https://www.omnicalculator.com/sports/calories-burned-by-heart-rate 

Looks quite pro-ish and might be an idea to implement its formula inside MTB.
